### PR TITLE
plugin support, design refresh, bug fixes, code cleanup, etc

### PR DESCRIPTION
**Pull request by rsjudka**
this shouldnt have gotten so big but oh :whale:

### CODE REVIEW

File: plugins/brightness/official_rpi/official_rpi.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#include <QDebug>
+
+#include "official_rpi.hpp"
+
+OfficialRPi::OfficialRPi() : brightness_attribute(this->PATH)
+{
+    qDebug() << "[Dash][OfficialRPi] brightness attribute exists: " << this->brightness_attribute.exists(); // temp
```
Comment (dannybarake) [inline plugins/brightness/official_rpi/official_rpi.cpp]: debug rules can be removed i guess

File: src/app/config.cpp [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
     : QObject(qApp),
       openauto_config(std::make_shared<openauto::configuration::Configuration>()),
       openauto_button_codes(openauto_config->getButtonCodes()),
-      ia_config(QSettings::IniFormat, QSettings::UserScope, "ia")
+      ia_config(QSettings::IniFormat, QSettings::UserScope, "dash")
 {
+    this->load_brightness_plugins();
+    this->brightness_active_plugin = new QPluginLoader(this);
+
     this->volume = this->ia_config.value("volume", 50).toInt();
```
Comment (dannybarake) [inline src/app/config.cpp]: maybe you can change ia_config into dash_config?

File: src/app/pages/vehicle.cpp [hunk 3]
```diff
@@ -71,6 +73,121 @@ QString Gauge::null_value()
     return null_str;
 }
 
+VehiclePage::VehiclePage(QWidget *parent) : QTabWidget(parent)
+{
+    this->tabBar()->setFont(Theme::font_16);
```
Comment (dannybarake) [inline src/app/pages/vehicle.cpp]: default fontsize is 14 i guess... Should be the same here?

**Review by dannybarake — COMMENTED** [review]
@rsjudka 3 small comments, do you see them now?

**Review by robert5974 — COMMENTED** [review]
Was the commit a question?  Is what is here missing something else?

**Review by robert5974 — COMMENTED** [review]
Was this commit a question? Is something missing?

File: src/app/config.cpp (hunk 2 above)
Comment (rsjudka) [reply src/app/config.cpp]: ah good catch! thought i removed them all lol

File: src/app/pages/vehicle.cpp (hunk 3 above)
Comment (rsjudka) [reply src/app/pages/vehicle.cpp]: yeah i originally kept them bigger, but i just tried out 14 and i like it better

(just a note if you're using the pi there's a bug where not all the tabs will have the correct size.. I'm sure you noticed sometimes the font is suuuper tiny lol)

**Review by robert5974 — COMMENTED** [review]
Was there something more that should have been here?  Just curious due to the description of commit.

File: src/app/pages/vehicle.cpp (hunk 3 above)
Comment (rsjudka) [reply src/app/pages/vehicle.cpp]: you actually gave me a little aha moment when you said default fontsize which fixed the qt bug! basically i set the default font for the entire app so tabs should be normally sized now!

**Comment by dannybarake** [comment]
From my point of view it looks good, ready to merge 👍

**Review by icecube45 — COMMENTED** [review]
My only question is - what is this blank rpi.sh? - otherwise looks great.

**Comment by rsjudka** [comment]
> My only question is - what is this blank rpi.sh? - otherwise looks great.

just made it an executable (`chmod +x`) weird that it says its empty

**Comment by icecube45** [comment]
Ah ok, looks great then.

**Comment by ZeroErrors** [comment]
I've swapped to testing this branch and seems to be working well.
The only issue I've run into is that modified settings won't get saved unless you use the controls bar to exit dash or restart the system.

I guess simply adding a save button on the settings page would work, or maybe saving settings when changing off of the settings page?

**Comment by icecube45** [comment]
I thought settings were to be saved on every change, but that might no longer be the case?
@ZeroErrors is that just on the settings page, or also on the per tab settings?

**Comment by ZeroErrors** [comment]
@icecube45 just on the settings page.
The settings accessible from the Android Auto page have their own save button but that's only for the `openauto.ini` from the looks.

Based on what I can find (still learning the codebase) in the code, the old code was saving settings every 10 seconds using a `QTimer`
https://github.com/openDsh/dash/blob/b108083226b1ec5bb58d3e5786b79e3d0127d573/src/app/config.cpp#L48-L50
but that code was removed in this PR https://github.com/openDsh/dash/pull/34/files#diff-5205fd2bb397aca6149095bef3b5ade9214c93f6cba53ee651bdabe8cb9c68abL48-L50

I'm in favor of using something other than a timer for saving anyway since they are very infrequently changed.

**Comment by icecube45** [comment]
Yup, you're right. Looks like the only save calls are done on control bar buttons now. Definitely should change that.

**Comment by rsjudka** [comment]
ah yeah totally forgot about the save button! I was meaning to do the save in the destructor, so basically the config will get saved on closing dash (regardless of how thats initiated)... except now im think the only other way to close dash is by killing it somehow and i dont wanna deal with handling signals lol

so ill definitely need to rethink this, but ill leave that to my next feature (which will basically make the config an accessible "api")

i really dont want a save button tho, trying to keep the config stuff transparent to the user

**Comment by icecube45** [comment]
@rsjudka only issue with this then - how do I save my option of not showing the control bar?

**Comment by rsjudka** [comment]
@icecube45 ah shoot yeah didn't think of that, I'll throw a save button in the settings page for now

Out of curiosity how do you close dash and/or shutdown the pi then?

**Comment by icecube45** [comment]
My setup is fully integrated into the car, so when my car's ignition state goes low, it triggers a graceful shutdown of the pi (after a 4 minute buffer period to account for quick stops).

So, a graceful killing of dash.

**Pull request merged**